tests/fault_tolerance/deploy/parse_results.py

```diff
@@ -90,6 +90,90 @@ def parse_test_log(
     return startup_time, failure_info
 
 
+def parse_timestamp(timestamp_str: str) -> Optional[datetime]:
+    """
+    Robustly parse timestamp with multiple format attempts.
+
+    Args:
+        timestamp_str: Timestamp string to parse
+
+    Returns:
+        datetime object or None if parsing fails
+    """
+    # List of common timestamp formats to try
+    timestamp_formats = [
+        "%Y-%m-%dT%H:%M:%S.%fZ",  # Full format with microseconds and Z
```

Comment:
question: which of the following are used? do we need to support all of them - just wondering - which ones we produce and if we can simplify